profilegen/src/profile_generator.rb

```diff
@@ -0,0 +1,238 @@
+  #
+  def generate_profile_gemspec
+    @profile_version = get_profile_version
+    file = get_spec_file
+    file.write(get_renderer(ProfileTemplates.spec_template))
+  end
+
+  #
+  # Method to generate the License file. Copies
+  # the license file from resources folder to destination
+  #
+  def generate_license_and_rakefile
+    profile_folder = "#{@output_dir}/#{@profile_name}/"
+    copy_file './res/LICENSE.txt', profile_folder
+    copy_file './res/Rakefile', profile_folder
+  end
+
+  #
+  # Method to generate the utils files.
+  # Copies the configurable.rb & default.rb files from the
+  # resources folder to destination.
+  #
+  def generate_utils
+    utils_folder = "#{@output_dir}/#{@profile_name}/lib/utils/"
+    copy_file './res/configurable.rb', utils_folder
+    copy_file './res/default.rb', utils_folder
+  end
+
+  def copy_file(source, destination)
+    FileUtils.cp source, destination
+  end
+
+  #
+  # Method to generate the individual module files per RP.
+  # For each RP in the profile:
+  #        1. Load the main file
+  #        2. Generate the list of operations
+  #        3. Generate the list of models
+  #        4. Generate the module file
+  #
+  def generate_modules
+    @resource_types.each do |resource_type_name, resource_type_version|
+      base_file_path = @dir_metadata[resource_type_name]['path'] + get_sdk_path + "azure_mgmt_#{get_ruby_specific_resource_type_name(resource_type_name).downcase}.rb"
+      require base_file_path
+      @module_require = @dir_metadata[resource_type_name]['module_require']
+      @spec_includes << @module_require
+      @module_name    = get_module_name(resource_type_name)
+      @class_name     = get_ruby_specific_resource_type_name(resource_type_name)
+      @module_class_mapper << { 'module_name': @module_name, 'class_name': @class_name }
+      generate_operation_types(resource_type_name)
+      generate_model_types(resource_type_name)
+      file = get_module_file resource_type_name
+      file.write(get_renderer(ProfileTemplates.module_template))
+    end
+  end
+
+  #
+  # Method to generate the client file
+  # Gets the client template and renders it
+  #
+  def generate_client
+    file = get_client_file
+    file.write(get_renderer(ProfileTemplates.client_template))
+  end
+
+  #
+  # Method to get renderer
+  #
+  # @param template for which the renderer will be created
+  #
+  def get_renderer(template)
+    renderer = ERB.new(template, 0, '-%>')
+    renderer.result(get_binding)
+  end
+
+  #
+  # Method to generate operations
+  #
+  # @param resource_type_name name of the resource for which the
+  # operations to be generated
+  #
+  def generate_operation_types(resource_type_name)
+    obj = Module.const_get(@dir_metadata[resource_type_name]['namespace'])
+    operations = []
+    obj.constants.select do |const_name|
+      if((obj.const_get(const_name).instance_of?Class) && !const_name.to_s.end_with?("ManagementClient"))
```

Comment:
please leave a space between `instance_of?` & `Class` to increase readability or use `()`